uefi/src/proto/device_path/build.rs

```diff
@@ -83,8 +83,11 @@ impl<'a> DevicePathBuilder<'a> {
     }
 
     /// Create a builder backed by a `Vec`.
+    ///
+    /// The `Vec` is cleared before use.
     #[cfg(feature = "alloc")]
```

Comment:
Why doesn't the method create the vector itself now?

Reply:
For performance.  [`clear`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.clear) doesn't free any memory, it just resets the initialized length. So you could use [`Vec::with_capacity`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity) to preallocate before building the device path.